GDJS/scripts/lib/runtime-files-list.js

```diff
@@ -45,6 +44,30 @@ const untransformedPaths = [
   'Extensions/JsExtensionTypes.flow.js',
 ].map((untransformedPath) => path.resolve(gdevelopRootPath, untransformedPath));
 
+// Files under these paths (relative to the GDevelop root path) will be completely
+// ignored. This is useful for example for ESM source files, where only the
+// entrypoint should be compiled.
+const ignoredPaths = [
+  // TileMapHelper source files:
+  'Extensions/TileMap/TileMapHelper/src',
+].map((untransformedPath) => path.resolve(gdevelopRootPath, untransformedPath));
+
+/**
+ * The list of modules that are not namespaced typescript but ES Modules to be bundled as a namespace.
+ * The key is the path to the module relative to the GDevelop root path, and the value is the name of
+ * the namespace where to put all exports of the ES Module.
+ * @type {Record<string, string>}
+ */
+const ESModulesList = Object.fromEntries(
```

Comment:
"esModuleAndNamespaces"'? Ideally the name would avoid the need for the comment that describe the key and value. Better: use objects with properties?